jqqb/rule.py

```diff
@@ -1,114 +1,60 @@
+            operator=Operator.get_operator(parsed_rule_json["operator"]),
+            inputs=[
+                Input.create_input_from_json(
+                    input_json=parsed_rule_input_json
                 )
-                else result
-            )
-            if result is None:
-                break
-        if isinstance(result, list):
-            return list(map(lambda x: self.typecast_value(x), result))
-        else:
-            return self.typecast_value(result)
-
-    def get_value(self):
-        if isinstance(self.value, list):
-            return list(map(lambda x: self.typecast_value(x), self.value))
-        return self.typecast_value(self.value)
-
-    def typecast_value(self, value_to_cast):
-        if value_to_cast is None:
-            return None
-
-        if self.type == "string":
-            return str(value_to_cast)
-        elif self.type == "integer":
-            return int(value_to_cast)
-        elif self.type == "double":
-            return float(value_to_cast)
-        elif self.type == "datetime":
-            return (
-                datetime.strptime(value_to_cast, "%Y-%m-%dT%H:%M:%S.%fZ")
-                if isinstance(value_to_cast, str)
-                else value_to_cast
-            )
-        elif self.type == "date":
-            return (
-                datetime.strptime(value_to_cast, "%Y-%m-%d")
-                if isinstance(value_to_cast, str)
-                else value_to_cast
-            )
-        elif self.type == "time":
-            return (
-                timeparse(value_to_cast)
-                if isinstance(value_to_cast, str)
-                else value_to_cast
-            )
-        elif self.type == "boolean":
-            if isinstance(value_to_cast, str):
-                return value_to_cast.lower() in [
-                    "true",
-                    "1",
-                    "t",
-                    "y",
-                    "yes",
-                    "yeah",
-                    "yup",
-                    "certainly",
-                    "uh-huh",
-                ]
-        elif self.type == "list":
-            ...
-        return value_to_cast
+                for parsed_rule_input_json
+                in parsed_rule_json["inputs"]
+            ]
+        )
+
+    def _get_input_values(self, object: dict) -> list:
+        return [input.get_value(object=object) for input in self.inputs]
+
+    def evaluate(self, object: dict) -> bool:
+        input_values = self._get_input_values(object)
+        return (
+            False
+            if Input.MissingKey in input_values
+            else self.operator(*input_values)
+        )
+
+    def get_phrase(self, object: dict) -> str:
+        input_phrases = [
+            str(input.get_value(object=object)) for input in self.inputs
+        ]
+        return (
+            f"{Operator.get_operator_phrase(operator=self.operator)}("
+            f"{', '.join(input_phrases)})"
+        )
```

Comment:
`get_proposition`, `get_expression` or `get_sentence`

Reply:
to be more precise, `predicate` or `statement` should be prefered terms

Reply:
to be more precise, `predicate` or `statement` should be prefered terms

Reply:
En effet, je me suis planté sur la traduction.
`proposition` me paraît pas mal, mais si tu préfères `predicate` ou `statement`, je te laisse décider lequel choisir. ;)

Reply:
Allez vas pour `predicate` (réf. https://en.wikipedia.org/wiki/First-order_logic).